src/doom/d_main.c

```diff
@@ -1081,6 +1085,21 @@ static void InitGameVersion(void)
                     }
                 }
             }
+
+            // detect v1.0/v1.1 from missing D_INTROA
+            if (gameversion == exe_doom_1_2 && W_CheckNumForName("D_INTROA") < 0)
+            {
+                lumpindex_t demo1 = W_CheckNumForName("demo1");
+
+                gameversion = exe_doom_1_1;
+
+                // detect v1.0 from length of demo1 lump
+                if (demo1 && W_LumpLength(demo1) < 10 * 1024)
```

Comment:
This should probably be `if (demo1 > -1`

Reply:
Fixed in https://github.com/chocolate-doom/chocolate-doom/pull/1207/commits/fb10e906055fb6464aaf74f0d2ac0c949a541773 .